llvm/include/llvm/MC/MCSubtargetInfo.h

```diff
@@ -93,6 +94,10 @@ class MCSubtargetInfo {
   FeatureBitset FeatureBits;           // Feature bits for current CPU + FS
   std::string FeatureString;           // Feature string
 
+  // General purpose name table, currently only used to store MCSchedClassDesc
+  // names when assertions are enabled.
+  StringTable NameTable;
```

Comment:
can we guard this def as well with `if !defined(NDEBUG) || defined(LLVM_ENABLE_DUMP)`? And in the constructor, assert that the `NT` passed is empty() in non-assert builds optionally?